x-pack/legacy/plugins/siem/public/components/ml_popover/api.tsx

```diff
@@ -141,52 +118,44 @@ export const stopDatafeeds = async (
   datafeedIds: string[],
   headers: Record<string, string | undefined>
 ): Promise<[StopDatafeedResponse, CloseJobsResponse]> => {
-  try {
-    const stopDatafeedsResponse = await fetch(
-      `${chrome.getBasePath()}/api/ml/jobs/stop_datafeeds`,
-      {
-        method: 'POST',
-        credentials: 'same-origin',
-        body: JSON.stringify({
-          datafeedIds,
-        }),
-        headers: {
-          'kbn-system-api': 'true',
-          'content-type': 'application/json',
-          'kbn-xsrf': chrome.getXsrfToken(),
-          ...headers,
-        },
-      }
-    );
-
-    await throwIfNotOk(stopDatafeedsResponse);
-    const stopDatafeedsResponseJson = await stopDatafeedsResponse.json();
-
-    const datafeedPrefix = 'datafeed-';
-    const closeJobsResponse = await fetch(`${chrome.getBasePath()}/api/ml/jobs/close_jobs`, {
-      method: 'POST',
-      credentials: 'same-origin',
-      body: JSON.stringify({
-        jobIds: datafeedIds.map(dataFeedId =>
-          dataFeedId.startsWith(datafeedPrefix)
-            ? dataFeedId.substring(datafeedPrefix.length)
-            : dataFeedId
-        ),
-      }),
-      headers: {
-        'content-type': 'application/json',
-        'kbn-system-api': 'true',
-        'kbn-xsrf': chrome.getXsrfToken(),
-        ...headers,
-      },
-    });
-
-    await throwIfNotOk(stopDatafeedsResponseJson);
-    return [stopDatafeedsResponseJson, await closeJobsResponse.json()];
-  } catch (error) {
-    // TODO: Toaster error when this happens instead of returning empty data
-    return emptyStopDatafeeds;
-  }
+  const stopDatafeedsResponse = await fetch(`${chrome.getBasePath()}/api/ml/jobs/stop_datafeeds`, {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      datafeedIds,
+    }),
+    headers: {
+      'kbn-system-api': 'true',
+      'content-type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+      ...headers,
+    },
+  });
+
+  await throwIfNotOk(stopDatafeedsResponse);
+  const stopDatafeedsResponseJson = await stopDatafeedsResponse.json();
+
+  const datafeedPrefix = 'datafeed-';
+  const closeJobsResponse = await fetch(`${chrome.getBasePath()}/api/ml/jobs/close_jobs`, {
+    method: 'POST',
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      jobIds: datafeedIds.map(dataFeedId =>
+        dataFeedId.startsWith(datafeedPrefix)
+          ? dataFeedId.substring(datafeedPrefix.length)
+          : dataFeedId
+      ),
+    }),
+    headers: {
+      'content-type': 'application/json',
+      'kbn-system-api': 'true',
+      'kbn-xsrf': chrome.getXsrfToken(),
+      ...headers,
+    },
+  });
+
+  await throwIfNotOk(closeJobsResponse);
+  return [stopDatafeedsResponseJson, await closeJobsResponse.json()];
```

Comment:
This file cleaned up real well with the way you've structured you're error reporting -- very nice! 👏